### DISCUSSION TITLE
Generate a machine readeable list of builtins

### DISCUSSION THREAD

**Discussion by traxys** [nvimtools/none-ls.nvim#71] [Ideas]
As part of [nixvim](https://github.com/nix-community/nixvim/) we generate Nix options for each of the available none-ls sources (see [here](https://github.com/nix-community/nixvim/blob/main/plugins/none-ls/servers.nix)).

This creates a bit of a mismatch between nixvim users and the none-ls docs, as not all sources are implemented, and we often have many PRs to add the sources that people need.
Would it be possible to generate a list of sources in something like JSON that we would be able to consume to automatically generate our options?

**Comment by mochaaP** [comment]
Definitely. See <https://github.com/nvimtools/none-ls.nvim/blob/main/scripts/autogen.lua>
